### PR TITLE
#159965481 Implement CRUD Functions for Articles.

### PR DESCRIPTION
**Pull request by Mutuba**
## What does this PR do?
```
Implements list articles (get a list of all articles available)
Implements get a single article function by providing article slug
Implements update a single article by an owner of the article
Implements delete a single article by an owner of the article
```

## Description of Task to be completed?
```
Have CRUD functions for Authors Heaven working.

```

## How should this be manually tested?

```
Clone the repo, cd into ah-leagueOfLegends. pip install -r requirements.txt.
Run python manage.py runserver. Go to postman, register and login.
Create an article using the following endpoint using HTTP POST: http://127.0.0.1:8000/api/articles
Sample data: {
  "article": {
    "title": "How to train your dragon",
    "description": "Ever wonder how?",
    "body": "You have to believe",
    "tagList": ["reactjs", "angularjs", "dragons"]
  }
}

```
## Any background context you want to provide?

```
The application had only authentication (registration and login). 

```

## What are the relevant pivotal tracker stories?

[#159965481 ](link)

## Screenshots if available to support PR.

screen shot 2018-09-07 at 10 11 05
screen shot 2018-09-07 at 10 14 02

## Questions:

```
What do you think of the CRUD implementation?
What could I have done better?

```

### CODE REVIEW

**Comment by ghost** [comment]
@Mutuba Please resolve the conflicts

**Comment by Mutuba** [comment]
Have solved conflicts!

**Comment by ghost** [comment]
@Mutuba The code looks good but Travis build is failing

**Review by Quantum-35 — DISMISSED** [review]

**Review by ghost — DISMISSED** [review]

**Review by Quantum-35 — DISMISSED** [review]

**Review by ghost — DISMISSED** [review]

**Review by DerKip — DISMISSED** [review]
Good job. I like the crud implementation. A suggestion to make it better, would be adding the ```taglists\<tag>``` in the url to retrive related articles

File: authors/settings.py [hunk 1]
```diff
@@ -20,7 +20,8 @@
 # See https://docs.djangoproject.com/en/1.11/howto/deployment/checklist/
 
 # SECURITY WARNING: keep the secret key used in production secret!
-SECRET_KEY = os.getenv('SECRET_KEY')
+#SECRET_KEY = os.getenv('SECRET_KEY')
+SECRET_KEY = '7pgozr2jn7zs_o%i8id6=rddie!*0f0qy3$oy$(8231i^4*@u3'
```
Comment (WNjihia) [inline authors/settings.py]: @Mutuba it's not safe to leave the secret this vulnerable. he previous implementation was better 
`SECRET_KEY = os.getenv('SECRET_KEY')`

**Review by WNjihia — COMMENTED** [review]
@Mutuba ~I don't see the tests for this~

File: authors/settings.py (hunk 1 above)
Comment (Mutuba) [reply authors/settings.py]: I have updated the settings.py. Thank you for noting!

**Review by Teddykavooh — APPROVED** [review]
Commendable work.

**Review by Quantum-35 — APPROVED** [review]
Looks good to me

**Review by WNjihia — APPROVED** [review]

**Pull request merged**